netbox/utilities/forms/bulk_import.py

```diff
@@ -86,12 +93,33 @@ def _detect_format(self, data):
             'format': _('Unable to detect data format. Please specify.')
         })
 
-    def _clean_csv(self, data):
+    def _clean_csv(self, data, delimiter=CSVDelimiterChoices.AUTO):
         """
         Clean CSV-formatted data. The first row will be treated as column headers.
         """
+
+        if delimiter == CSVDelimiterChoices.AUTO:
+            # Determine the CSV dialect
+            try:
+                # This uses a rough heuristic to detect the CSV dialect. If the data is malformed, we'll fall back to
+                # the default Excel dialect. Note that delimiter can only be one character.
+                dialect = csv.Sniffer().sniff(
+                    data.strip(), delimiters=''.join([CSVDelimiterChoices.COMMA, CSVDelimiterChoices.SEMICOLON])
```

Comment:
Add CSVDelimiterChoices.TAB to this list. I've tried it locally, and it works fine for me. I know you're concerned that '\t' may be considered as two character by the Sniff internals, but trust me, it works. `'\t'` is turned into a single ascii code already by the Python interpreter's string parsing.